src/patches/slices_to_json.cpp

```diff
@@ -148,6 +155,8 @@ json slice_to_json(const SparseSlice& slice)
             json visual_array_cell = boundaries_to_array_edges_json(routing_cell);
             visual_array_cell["patch_type"] = "Ancilla";
             visual_array_cell["activity"] = json({});
+            add_routing_region_id(visual_array_cell, *routing_region.routing_region_id);
```

Comment:
`routing_region_id` is an optional, but you use the special value of -1 check whether it should be added or not. Perhaps you want to:
 1. Initialize to the default (i.e `std::optional<OpId> routing_region_id;`)
 2. not dereference here (i.e. remove the `*`) on the line above
 3. use `routing_region_id` has value in `add_routing_region_id`